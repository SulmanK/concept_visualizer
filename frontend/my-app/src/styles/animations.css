/* Animation Utilities */

/* Base transitions */
.transition-all {
  transition-property: all;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 300ms;
}

.transition-transform {
  transition-property: transform;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 300ms;
  /* Tell browser to optimize for transform changes */
  will-change: transform;
}

.transition-opacity {
  transition-property: opacity;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 300ms;
  /* Tell browser to optimize for opacity changes */
  will-change: opacity;
}

.transition-colors {
  transition-property: color, background-color, border-color,
    text-decoration-color, fill, stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 300ms;
}

/* Duration variations */
.duration-fast {
  transition-duration: 150ms;
}

.duration-normal {
  transition-duration: 300ms;
}

.duration-slow {
  transition-duration: 500ms;
}

/* Hover effects - optimized to use transform which is GPU accelerated */
.hover-lift {
  transition-property: transform, box-shadow;
  transition-duration: 300ms;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, box-shadow;
}

.hover-lift:hover {
  transform: translateY(-3px);
  box-shadow:
    0 10px 25px -5px rgba(79, 70, 229, 0.1),
    0 8px 10px -6px rgba(79, 70, 229, 0.05);
}

.hover-bright {
  transition-property: filter;
  transition-duration: 300ms;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
}

.hover-bright:hover {
  filter: brightness(1.05);
}

.hover-scale {
  transition-property: transform;
  transition-duration: 300ms;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform;
}

.hover-scale:hover {
  transform: scale(1.05);
}

/* Optimized entrance animations using transform and opacity */
.fade-in {
  animation: fade-in 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: opacity;
}

.slide-in-up {
  animation: slide-in-up 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, opacity;
}

.slide-in-down {
  animation: slide-in-down 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, opacity;
}

.slide-in-left {
  animation: slide-in-left 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, opacity;
}

.slide-in-right {
  animation: slide-in-right 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, opacity;
}

.scale-in {
  animation: scale-in 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  will-change: transform, opacity;
}

/* Performance-optimized keyframes using transform and opacity */
@keyframes fade-in {
  0% {
    opacity: 0;
    visibility: hidden;
  }
  1% {
    visibility: visible;
  }
  100% {
    opacity: 1;
    visibility: visible;
  }
}

@keyframes slide-in-up {
  0% {
    opacity: 0;
    visibility: hidden;
    transform: translateY(20px);
  }
  1% {
    visibility: visible;
  }
  100% {
    opacity: 1;
    visibility: visible;
    transform: translateY(0);
  }
}

@keyframes slide-in-down {
  0% {
    opacity: 0;
    transform: translateY(-20px);
  }
  100% {
    opacity: 1;
    transform: translateY(0);
  }
}

@keyframes slide-in-left {
  0% {
    opacity: 0;
    transform: translateX(-20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes slide-in-right {
  0% {
    opacity: 0;
    transform: translateX(20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes scale-in {
  0% {
    opacity: 0;
    transform: scale(0.95);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}

/* Special effect animations */
.pulse {
  animation: pulse 2s cubic-bezier(0.4, 0, 0.6, 1) infinite;
  will-change: opacity;
}

@keyframes pulse {
  0%,
  100% {
    opacity: 1;
  }
  50% {
    opacity: 0.7;
  }
}

.bounce {
  animation: bounce 1s infinite;
  will-change: transform;
}

@keyframes bounce {
  0%,
  100% {
    transform: translateY(0);
    animation-timing-function: cubic-bezier(0.8, 0, 1, 1);
  }
  50% {
    transform: translateY(-10px);
    animation-timing-function: cubic-bezier(0, 0, 0.2, 1);
  }
}

/* Focus animations - for form elements */
.focus-ring {
  transition: box-shadow 0.2s ease-in-out;
}

.focus-ring:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: 0 0 0 3px rgba(79, 70, 229, 0.45);
}

/* Performance optimized hover class for buttons */
.btn-hover {
  position: relative;
  overflow: hidden;
}

.btn-hover::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(255, 255, 255, 0.1);
  opacity: 0;
  transition: opacity 0.3s;
  pointer-events: none;
  will-change: opacity;
}

.btn-hover:hover::after {
  opacity: 1;
}

/* For mobile devices - reduce motion/animations to improve performance */
@media screen and (max-width: 768px) {
  .transition-all,
  .transition-transform,
  .transition-opacity,
  .transition-colors {
    transition-duration: 200ms; /* Faster transitions on mobile */
  }

  .hover-lift:hover {
    transform: translateY(-2px); /* Smaller transform on mobile */
  }

  .hover-scale:hover {
    transform: scale(1.03); /* Smaller scale on mobile */
  }

  /* Reduce animation duration on mobile */
  .fade-in,
  .slide-in-up,
  .slide-in-down,
  .slide-in-left,
  .slide-in-right,
  .scale-in {
    animation-duration: 0.3s;
  }
}

/* Reduced motion preferences */
@media (prefers-reduced-motion: reduce) {
  .transition-all,
  .transition-transform,
  .transition-opacity,
  .transition-colors,
  .hover-lift,
  .hover-bright,
  .hover-scale,
  .fade-in,
  .slide-in-up,
  .slide-in-down,
  .slide-in-left,
  .slide-in-right,
  .scale-in,
  .pulse,
  .bounce {
    transition: none !important;
    animation: none !important;
    transform: none !important;
  }
}

/* Prevent content flash during page transitions */
.page-transition-container {
  will-change: opacity, transform, visibility;
  backface-visibility: hidden;
  transform-style: preserve-3d;
  contain: content;
  opacity: 0;
  visibility: hidden;
}

.page-transition-container[style*="visibility: visible"] {
  opacity: 1;
}

/* Optimize animations for better performance */
.animated-transition {
  will-change: opacity, transform, visibility;
  backface-visibility: hidden;
  transform-style: preserve-3d;
  contain: content;
  opacity: 0;
  visibility: hidden;
  transition:
    opacity var(--duration) ease-out,
    visibility var(--duration) ease-out,
    transform var(--duration) ease-out;
}

.animated-transition[style*="visibility: visible"] {
  opacity: 1;
  visibility: visible;
}

/* Remove the initial page load classes since we're handling it differently now */
.initial-page-load,
.initial-page-load.loaded {
  display: none;
}

/* Update animation states to include visibility */
.fade-enter,
.fade-exit,
.slide-up-enter,
.slide-up-exit,
.slide-down-enter,
.slide-down-exit,
.slide-left-enter,
.slide-left-exit,
.slide-right-enter,
.slide-right-exit {
  will-change: opacity, transform, visibility;
  backface-visibility: hidden;
  transform-style: preserve-3d;
  contain: content;
  visibility: hidden;
  transition:
    opacity var(--duration) ease-out,
    visibility var(--duration) ease-out,
    transform var(--duration) ease-out;
}
