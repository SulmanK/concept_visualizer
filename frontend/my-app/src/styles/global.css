@import "./variables.css";

/* Tailwind directives */
@tailwind base;
@tailwind components;
@tailwind utilities;

/* CSS Reset to remove default margins */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html,
body {
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
}

body {
  font-family:
    "Montserrat",
    system-ui,
    -apple-system,
    sans-serif;
}

@layer base {
  body {
    @apply bg-gradient-to-br from-indigo-50 to-blue-100 min-h-screen;
    font-family:
      "Montserrat",
      system-ui,
      -apple-system,
      sans-serif;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    @apply font-bold text-indigo-900;
  }

  h1 {
    @apply text-3xl md:text-4xl;
  }

  h2 {
    @apply text-2xl md:text-3xl;
  }

  h3 {
    @apply text-xl md:text-2xl;
  }

  a {
    @apply text-primary hover:text-primary-dark transition-colors;
  }
}

@layer components {
  /* Buttons */
  .btn-primary {
    @apply bg-gradient-to-r from-primary to-primary-dark text-white px-6 py-3 rounded-lg shadow-modern hover:shadow-modern-hover transition-all duration-200 font-semibold;
  }

  .btn-secondary {
    @apply bg-gradient-to-r from-secondary to-secondary-dark text-white px-6 py-3 rounded-lg shadow-modern hover:shadow-modern-hover transition-all duration-200 font-semibold;
  }

  .btn-outline {
    @apply border-2 border-primary/50 bg-white/80 hover:bg-white text-primary px-6 py-3 rounded-lg shadow-sm hover:shadow-modern transition-all duration-200 font-semibold;
  }

  /* Cards */
  .card {
    @apply bg-white/90 backdrop-blur-sm rounded-lg shadow-modern border border-indigo-100 overflow-hidden;
  }

  .card-gradient {
    @apply bg-gradient-to-br from-indigo-50/90 to-white/90 backdrop-blur-sm rounded-lg shadow-modern border border-indigo-100 overflow-hidden;
  }

  .card-title {
    @apply text-xl font-semibold text-indigo-900 mb-6;
  }

  /* Nav links */
  .nav-link {
    @apply px-4 py-2 rounded-lg text-sm font-semibold transition-all duration-200;
  }

  .nav-link-active {
    @apply bg-gradient-to-r from-primary to-primary-dark text-white shadow-modern;
  }

  .nav-link-inactive {
    @apply text-indigo-700 hover:bg-indigo-50 hover:text-primary transition-colors;
  }

  /* Form elements */
  .input {
    @apply w-full px-4 py-3 rounded-lg border border-indigo-200 focus:ring-2 focus:ring-primary/30 focus:border-primary focus:outline-none transition-all duration-200;
  }

  .label {
    @apply block text-sm font-medium text-indigo-700 mb-2;
  }

  /* Helper text */
  .helper-text {
    @apply mt-2 text-xs text-gray-500;
  }

  /* Error messages */
  .error-message {
    @apply flex items-center p-4 rounded-lg text-sm bg-gradient-to-r from-red-50 to-indigo-50 text-red-600 border border-indigo-100 shadow-lg;
  }

  /* Gradient backgrounds */
  .bg-gradient-primary {
    @apply bg-gradient-to-r from-primary to-primary-dark;
  }

  .bg-gradient-secondary {
    @apply bg-gradient-to-r from-secondary to-secondary-dark;
  }

  .bg-gradient-page {
    @apply bg-gradient-to-br from-indigo-50 to-blue-100;
  }

  /* Gradient text */
  .gradient-text {
    @apply bg-gradient-to-r from-primary to-secondary bg-clip-text text-transparent;
  }

  /* Interactive elements animations */
  .hover-lift {
    @apply transition-transform duration-300 hover:-translate-y-1;
  }

  .hover-scale {
    @apply transition-transform duration-300 hover:scale-105;
  }

  .hover-bright {
    @apply transition-all duration-300 hover:brightness-110;
  }
}

/* Animations */
@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.spinner {
  animation: spin 1s linear infinite;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.fade-in {
  animation: fadeIn 0.5s ease-in-out;
}

@keyframes scaleIn {
  from {
    transform: scale(0.9);
    opacity: 0;
  }
  to {
    transform: scale(1);
    opacity: 1;
  }
}

.scale-in {
  animation: scaleIn 0.3s ease-out;
}
