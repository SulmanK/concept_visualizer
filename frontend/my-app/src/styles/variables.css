:root {
  /* Colors - Primary */
  --color-primary: #4f46e5;
  --color-primary-dark: #4338ca;
  --color-primary-light: #818cf8;

  /* Colors - Secondary */
  --color-secondary: #7e22ce;
  --color-secondary-dark: #6b21a8;
  --color-secondary-light: #a78bfa;

  /* Colors - Neutrals */
  --color-indigo-50: #eef2ff;
  --color-indigo-100: #e0e7ff;
  --color-indigo-200: #c7d2fe;
  --color-indigo-600: #4f46e5;
  --color-indigo-700: #4338ca;
  --color-indigo-900: #312e81;

  /* Colors - Accents */
  --color-blue-100: #dbeafe;
  --color-blue-400: #60a5fa;
  --color-blue-500: #3b82f6;

  /* Sizes */
  --spacing-1: 0.25rem;
  --spacing-2: 0.5rem;
  --spacing-3: 0.75rem;
  --spacing-4: 1rem;
  --spacing-6: 1.5rem;
  --spacing-8: 2rem;
  --spacing-12: 3rem;
  --spacing-16: 4rem;

  /* Font Sizes */
  --text-xs: 0.75rem;
  --text-sm: 0.875rem;
  --text-base: 1rem;
  --text-lg: 1.125rem;
  --text-xl: 1.25rem;
  --text-2xl: 1.5rem;
  --text-3xl: 1.875rem;
  --text-4xl: 2.25rem;

  /* Shadows */
  --shadow-sm: 0 1px 2px 0 rgba(0, 0, 0, 0.05);
  --shadow-md: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
    0 2px 4px -1px rgba(0, 0, 0, 0.06);
  --shadow-lg: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  --shadow-modern: 0 10px 25px -5px rgba(0, 0, 0, 0.05),
    0 8px 10px -6px rgba(0, 0, 0, 0.02);
  --shadow-modern-hover: 0 20px 35px -10px rgba(0, 0, 0, 0.1),
    0 10px 15px -5px rgba(0, 0, 0, 0.04);

  /* Border Radius */
  --radius-sm: 0.125rem;
  --radius-md: 0.375rem;
  --radius-lg: 0.5rem;
  --radius-xl: 0.75rem;
  --radius-2xl: 1rem;
  --radius-full: 9999px;

  /* Transitions */
  --transition-fast: 150ms;
  --transition-normal: 200ms;
  --transition-slow: 300ms;
}
