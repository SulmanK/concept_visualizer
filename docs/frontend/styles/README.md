# Styles

This directory contains global styles, variables, animations, and theme configurations for the Concept Visualizer application.

## Directory Structure

The styles are organized into the following files:

- **Global styles**: Application-wide styles and CSS resets
- **Variables**: SCSS variables for colors, spacing, fonts, etc.
- **Animations**: Reusable animations and transitions
- **Theme**: Theme configuration for styled components or MUI

## Key Features

- **Responsive Design**: Mobile-first approach with breakpoints for different screen sizes
- **Consistent Colors**: Predefined color palette for brand consistency
- **Typography System**: Standardized font sizes, weights, and line heights
- **Spacing System**: Consistent spacing variables for margins and padding
- **Animation Library**: Reusable animations for interactive elements

## Usage

### Importing Styles

```typescript
// Global styles are imported in the main App component
import './styles/global.scss';

// For specific component styles
import './styles/components/Button.scss';
```

### Theme Variables

The application uses CSS variables for theming, allowing for easy dark/light mode switching and consistent styling across components:

```css
:root {
  --primary-color: #3a86ff;
  --secondary-color: #ff006e;
  --background-color: #ffffff;
  --text-color: #333333;
  
  --spacing-xs: 4px;
  --spacing-sm: 8px;
  --spacing-md: 16px;
  --spacing-lg: 24px;
  --spacing-xl: 32px;
  
  --font-family: 'Inter', -apple-system, BlinkMacSystemFont, sans-serif;
  --font-size-sm: 14px;
  --font-size-md: 16px;
  --font-size-lg: 18px;
  --font-size-xl: 24px;
}

/* Dark mode theme */
@media (prefers-color-scheme: dark) {
  :root {
    --background-color: #121212;
    --text-color: #f5f5f5;
  }
}
```

### Animations

The application includes several reusable animations:

```css
@keyframes fadeIn {
  from { opacity: 0; }
  to { opacity: 1; }
}

.fade-in {
  animation: fadeIn 0.3s ease-in-out;
}

@keyframes slideIn {
  from { transform: translateY(20px); opacity: 0; }
  to { transform: translateY(0); opacity: 1; }
}

.slide-in {
  animation: slideIn 0.4s ease-out;
}
```

## Best Practices

1. **Use Variables**: Always use the predefined variables for colors, spacing, and typography
2. **Mobile-First**: Start with mobile styles and use media queries for larger screens
3. **Component Encapsulation**: Keep component styles scoped to prevent leakage
4. **Consistent Naming**: Follow BEM (Block Element Modifier) naming convention
5. **Performance**: Minimize nested selectors and avoid expensive CSS properties 