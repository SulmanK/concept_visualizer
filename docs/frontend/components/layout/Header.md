# Header Component

The `Header` component serves as the main navigation bar at the top of the application, providing access to primary navigation links, user account controls, and branding.

## Overview

The Header is a fixed component that appears on all pages, providing consistent navigation across the application. It includes the application logo, main navigation links, and user-related controls like profile and authentication options.

## Usage

```tsx
import { Header } from "components/layout/Header";

function App() {
  return (
    <div className="app">
      <Header />
      <main>{/* Page content */}</main>
      <Footer />
    </div>
  );
}
```

## Props

| Prop          | Type      | Default | Description                                             |
| ------------- | --------- | ------- | ------------------------------------------------------- |
| `className`   | `string`  | `''`    | Additional CSS class to apply to the header             |
| `transparent` | `boolean` | `false` | Whether the header should have a transparent background |

## Implementation Details

```tsx
import React, { useState } from "react";
import {
  AppBar,
  Toolbar,
  Typography,
  Button,
  IconButton,
  Box,
  Menu,
  MenuItem,
  Container,
  useScrollTrigger,
  useMediaQuery,
} from "@mui/material";
import { Menu as MenuIcon, AccountCircle } from "@mui/icons-material";
import { useNavigate } from "react-router-dom";
import { useTheme } from "@mui/material/styles";
import { useAuth } from "contexts/AuthContext";
import logoImage from "assets/logos/logo.svg";

interface HeaderProps {
  className?: string;
  transparent?: boolean;
}

export function Header({ className = "", transparent = false }: HeaderProps) {
  const theme = useTheme();
  const navigate = useNavigate();
  const { user, signOut } = useAuth();
  const isMobile = useMediaQuery(theme.breakpoints.down("md"));

  // Change header appearance on scroll
  const scrollTrigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 50,
  });

  // Mobile menu state
  const [mobileMenuAnchor, setMobileMenuAnchor] = useState<null | HTMLElement>(
    null,
  );
  const isMobileMenuOpen = Boolean(mobileMenuAnchor);

  // User menu state
  const [userMenuAnchor, setUserMenuAnchor] = useState<null | HTMLElement>(
    null,
  );
  const isUserMenuOpen = Boolean(userMenuAnchor);

  // Handle mobile menu
  const handleMobileMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
    setMobileMenuAnchor(event.currentTarget);
  };

  const handleMobileMenuClose = () => {
    setMobileMenuAnchor(null);
  };

  // Handle user menu
  const handleUserMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
    setUserMenuAnchor(event.currentTarget);
  };

  const handleUserMenuClose = () => {
    setUserMenuAnchor(null);
  };

  // Navigation handlers
  const navigateTo = (path: string) => {
    navigate(path);
    handleMobileMenuClose();
    handleUserMenuClose();
  };

  const handleSignOut = () => {
    signOut();
    handleUserMenuClose();
    navigate("/");
  };

  return (
    <AppBar
      position="fixed"
      className={className}
      elevation={scrollTrigger || !transparent ? 4 : 0}
      sx={{
        backgroundColor:
          transparent && !scrollTrigger
            ? "transparent"
            : theme.palette.background.paper,
        transition: theme.transitions.create([
          "background-color",
          "box-shadow",
        ]),
      }}
    >
      <Container maxWidth="lg">
        <Toolbar disableGutters>
          {/* Logo */}
          <Box
            sx={{
              display: "flex",
              alignItems: "center",
              flexGrow: { xs: 1, md: 0 },
            }}
          >
            <img
              src={logoImage}
              alt="Concept Visualizer Logo"
              height={40}
              onClick={() => navigateTo("/")}
              style={{ cursor: "pointer" }}
            />
            <Typography
              variant="h6"
              sx={{
                ml: 1,
                display: { xs: "none", sm: "block" },
                color:
                  transparent && !scrollTrigger
                    ? theme.palette.common.white
                    : theme.palette.text.primary,
              }}
            >
              Concept Visualizer
            </Typography>
          </Box>

          {/* Desktop Navigation */}
          {!isMobile && (
            <Box
              sx={{ display: "flex", flexGrow: 1, justifyContent: "center" }}
            >
              <Button
                color="inherit"
                onClick={() => navigateTo("/")}
                sx={{
                  color:
                    transparent && !scrollTrigger
                      ? theme.palette.common.white
                      : theme.palette.text.primary,
                }}
              >
                Home
              </Button>
              <Button
                color="inherit"
                onClick={() => navigateTo("/concepts")}
                sx={{
                  color:
                    transparent && !scrollTrigger
                      ? theme.palette.common.white
                      : theme.palette.text.primary,
                }}
              >
                My Concepts
              </Button>
            </Box>
          )}

          {/* User Actions */}
          <Box sx={{ display: "flex" }}>
            {user ? (
              <>
                <IconButton
                  edge="end"
                  aria-label="account of current user"
                  aria-controls="user-menu"
                  aria-haspopup="true"
                  onClick={handleUserMenuOpen}
                  color="inherit"
                  sx={{
                    color:
                      transparent && !scrollTrigger
                        ? theme.palette.common.white
                        : theme.palette.text.primary,
                  }}
                >
                  <AccountCircle />
                </IconButton>
                <Menu
                  id="user-menu"
                  anchorEl={userMenuAnchor}
                  keepMounted
                  open={isUserMenuOpen}
                  onClose={handleUserMenuClose}
                >
                  <MenuItem onClick={() => navigateTo("/profile")}>
                    Profile
                  </MenuItem>
                  <MenuItem onClick={() => navigateTo("/settings")}>
                    Settings
                  </MenuItem>
                  <MenuItem onClick={handleSignOut}>Sign Out</MenuItem>
                </Menu>
              </>
            ) : (
              <Button
                color="primary"
                variant="contained"
                onClick={() => navigateTo("/login")}
              >
                Sign In
              </Button>
            )}

            {/* Mobile Menu Button */}
            {isMobile && (
              <>
                <IconButton
                  edge="start"
                  color="inherit"
                  aria-label="menu"
                  aria-controls="mobile-menu"
                  aria-haspopup="true"
                  onClick={handleMobileMenuOpen}
                  sx={{
                    ml: 1,
                    color:
                      transparent && !scrollTrigger
                        ? theme.palette.common.white
                        : theme.palette.text.primary,
                  }}
                >
                  <MenuIcon />
                </IconButton>
                <Menu
                  id="mobile-menu"
                  anchorEl={mobileMenuAnchor}
                  keepMounted
                  open={isMobileMenuOpen}
                  onClose={handleMobileMenuClose}
                >
                  <MenuItem onClick={() => navigateTo("/")}>Home</MenuItem>
                  <MenuItem onClick={() => navigateTo("/concepts")}>
                    My Concepts
                  </MenuItem>
                </Menu>
              </>
            )}
          </Box>
        </Toolbar>
      </Container>
    </AppBar>
  );
}
```

## Features

- **Responsive Design**: Adapts to different screen sizes, with a hamburger menu for mobile devices
- **Navigation**: Provides access to main application sections
- **User Authentication**: Displays sign-in button or user menu based on authentication state
- **Branding**: Prominently displays application logo and name
- **Transparency Option**: Can be transparent for use on hero sections, gradually becoming opaque on scroll
- **Scroll Behavior**: Changes appearance when scrolling down the page

## Accessibility

- Proper ARIA attributes for interactive elements
- Keyboard navigation support for menus
- Sufficient color contrast for text readability
- Proper alt text for logo image

## Customization

The Header can be customized through:

1. **Props**: Using the `transparent` prop to control background transparency
2. **CSS**: Using the `className` prop to apply custom styles
3. **Theme**: Customizing appearance through the Material-UI theme
4. **Extension**: Creating a wrapped version with additional navigation items
