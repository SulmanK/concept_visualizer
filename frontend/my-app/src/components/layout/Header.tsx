import React from 'react';
import { Link } from 'react-router-dom';
import styles from './header.module.css';

export interface HeaderProps {
  /**
   * The current active route
   */
  activeRoute?: string;
}

/**
 * Application header with navigation using modular CSS
 */
export const Header: React.FC<HeaderProps> = ({ activeRoute = '/' }) => {
  // Consider both '/' and '/create' as the create route since they show the same component
  const isCreateRoute = activeRoute === '/' || activeRoute === '/create';
  
  return (
    <header className={styles.headerContainer}>
      <div className={styles.innerContainer}>
        <div className={styles.headerRow}>
          {/* Logo and title */}
          <div className={styles.logoWrapper}>
            <Link to="/" className={styles.logoLink}>
              <div className={styles.logoCircle}>
                CV
              </div>
              <h1 className={styles.titleText}>
                Concept Visualizer
              </h1>
            </Link>
          </div>
          
          {/* Navigation */}
          <nav className={styles.navContainer}>
            <Link 
              to="/create" 
              className={isCreateRoute ? styles.activeNavLink : styles.inactiveNavLink}
            >
              <span className={styles.navIcon}>âœ¨</span>Create
            </Link>
            
            <Link 
              to="/refine" 
              className={activeRoute === '/refine' ? styles.activeNavLink : styles.inactiveNavLink}
            >
              <span className={styles.navIcon}>ðŸ”„</span>Refine
            </Link>
          </nav>
        </div>
      </div>
    </header>
  );
}; 