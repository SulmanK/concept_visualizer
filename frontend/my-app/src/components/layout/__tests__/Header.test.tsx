import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Header } from '../Header';
import styles from '../header.module.css';

// Helper function to render with router
const renderWithRouter = (ui: React.ReactElement, { route = '/' } = {}) => {
  return render(
    <MemoryRouter initialEntries={[route]}>
      {ui}
    </MemoryRouter>
  );
};

describe('Header Component', () => {
  // Basic rendering tests
  test('renders header with logo and title', () => {
    renderWithRouter(<Header />);
    
    // Check for logo text
    const logo = screen.getByText('CV');
    expect(logo).toBeInTheDocument();
    
    // Check for title
    const title = screen.getByText('Concept Visualizer');
    expect(title).toBeInTheDocument();
  });
  
  // Navigation tests
  test('renders navigation links', () => {
    renderWithRouter(<Header />);
    
    // Check for navigation links
    const createLink = screen.getByText(/Create/i);
    const refineLink = screen.getByText(/Refine/i);
    
    expect(createLink).toBeInTheDocument();
    expect(refineLink).toBeInTheDocument();
  });
  
  // Active route tests
  test('highlights Create link when on create route', () => {
    renderWithRouter(<Header activeRoute="/create" />);
    
    const createLink = screen.getByText(/Create/i).closest('a');
    const refineLink = screen.getByText(/Refine/i).closest('a');
    
    // Testing active/inactive CSS classes
    expect(createLink).toHaveAttribute('href', '/create');
    expect(refineLink).toHaveAttribute('href', '/refine');
    
    // Test CSS classes instead of inline styles
    expect(createLink).toHaveClass(styles.activeNavLink);
    expect(refineLink).toHaveClass(styles.inactiveNavLink);
  });
  
  test('highlights Refine link when on refine route', () => {
    renderWithRouter(<Header activeRoute="/refine" />);
    
    const createLink = screen.getByText(/Create/i).closest('a');
    const refineLink = screen.getByText(/Refine/i).closest('a');
    
    // Check that correct link is highlighted via CSS classes
    expect(refineLink).toHaveClass(styles.activeNavLink);
    expect(createLink).toHaveClass(styles.inactiveNavLink);
  });
  
  // Home route test
  test('treats root route as create route', () => {
    renderWithRouter(<Header activeRoute="/" />);
    
    const createLink = screen.getByText(/Create/i).closest('a');
    
    // Check that Create is highlighted on home route
    expect(createLink).toHaveClass(styles.activeNavLink);
  });
  
  // Snapshot tests
  describe('Snapshots', () => {
    test('default header snapshot', () => {
      const { container } = renderWithRouter(<Header />);
      expect(container.firstChild).toMatchSnapshot();
    });
    
    test('header with create route active snapshot', () => {
      const { container } = renderWithRouter(<Header activeRoute="/create" />);
      expect(container.firstChild).toMatchSnapshot();
    });
    
    test('header with refine route active snapshot', () => {
      const { container } = renderWithRouter(<Header activeRoute="/refine" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
}); 