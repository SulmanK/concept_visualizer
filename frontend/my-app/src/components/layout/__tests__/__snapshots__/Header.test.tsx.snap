// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > Snapshots > default header snapshot 1`] = `
<header
  class="_headerContainer_5ee9af"
>
  <div
    class="_innerContainer_5ee9af"
  >
    <div
      class="_headerRow_5ee9af"
    >
      <div
        class="_logoWrapper_5ee9af"
      >
        <a
          class="_logoLink_5ee9af"
          href="/"
        >
          <div
            class="_logoCircle_5ee9af"
          >
            CV
          </div>
          <h1
            class="_titleText_5ee9af"
          >
            Concept Visualizer
          </h1>
        </a>
      </div>
      <nav
        class="_navContainer_5ee9af"
      >
        <a
          class="_activeNavLink_5ee9af"
          href="/create"
        >
          <span
            class="_navIcon_5ee9af"
          >
            âœ¨
          </span>
          Create
        </a>
        <a
          class="_inactiveNavLink_5ee9af"
          href="/refine"
        >
          <span
            class="_navIcon_5ee9af"
          >
            ðŸ”„
          </span>
          Refine
        </a>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header Component > Snapshots > header with create route active snapshot 1`] = `
<header
  class="_headerContainer_5ee9af"
>
  <div
    class="_innerContainer_5ee9af"
  >
    <div
      class="_headerRow_5ee9af"
    >
      <div
        class="_logoWrapper_5ee9af"
      >
        <a
          class="_logoLink_5ee9af"
          href="/"
        >
          <div
            class="_logoCircle_5ee9af"
          >
            CV
          </div>
          <h1
            class="_titleText_5ee9af"
          >
            Concept Visualizer
          </h1>
        </a>
      </div>
      <nav
        class="_navContainer_5ee9af"
      >
        <a
          class="_activeNavLink_5ee9af"
          href="/create"
        >
          <span
            class="_navIcon_5ee9af"
          >
            âœ¨
          </span>
          Create
        </a>
        <a
          class="_inactiveNavLink_5ee9af"
          href="/refine"
        >
          <span
            class="_navIcon_5ee9af"
          >
            ðŸ”„
          </span>
          Refine
        </a>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header Component > Snapshots > header with refine route active snapshot 1`] = `
<header
  class="_headerContainer_5ee9af"
>
  <div
    class="_innerContainer_5ee9af"
  >
    <div
      class="_headerRow_5ee9af"
    >
      <div
        class="_logoWrapper_5ee9af"
      >
        <a
          class="_logoLink_5ee9af"
          href="/"
        >
          <div
            class="_logoCircle_5ee9af"
          >
            CV
          </div>
          <h1
            class="_titleText_5ee9af"
          >
            Concept Visualizer
          </h1>
        </a>
      </div>
      <nav
        class="_navContainer_5ee9af"
      >
        <a
          class="_inactiveNavLink_5ee9af"
          href="/create"
        >
          <span
            class="_navIcon_5ee9af"
          >
            âœ¨
          </span>
          Create
        </a>
        <a
          class="_activeNavLink_5ee9af"
          href="/refine"
        >
          <span
            class="_navIcon_5ee9af"
          >
            ðŸ”„
          </span>
          Refine
        </a>
      </nav>
    </div>
  </div>
</header>
`;
